### PR TITLE
Update docs to include SparkStreamingDataSet and ManagedTableDataSet

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by noklam** [comment]
As usual, never underestimate the time to fix an indentation error with Sphinx, but here we are! Big thanks to @AhdraMeraliQB !

File: setup.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def _collect_requirements(requires):
         "sphinxcontrib-mermaid~=0.7.1",
         "myst-parser~=1.0.0",
         "Jinja2<3.1.0",
-        "kedro-datasets[all]~=1.4.0",
+        # "kedro-datasets[all]~=1.4.0",
```
Comment (stichbury) [inline setup.py:107]: Is there a reason you've retained the old line (leaving it commented) when replacing it?

**Review by stichbury — APPROVED** [review]
LGTM 🏆 and yes, Sphinx errors :trollface:

**Review by astrojuanlu — CHANGES_REQUESTED** [review]

**Comment by noklam** [comment]
It's very hard to debug since the build keep failing, I add a pip list so it's easier to know if it is installing the correction version of `kedro-datasets`

File: .readthedocs.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ build:
     post_create_environment:
       - npm install -g @mermaid-js/mermaid-cli
     pre_build:
+      - pip list
```
Comment (astrojuanlu) [inline .readthedocs.yml:19]: If we're to have this, let's make it a `pip freeze` instead so we can copy-paste the output to a `requirements.txt`

**Review by astrojuanlu — APPROVED** [review]
Approved when/if the RTD build passes 😄

**Comment by astrojuanlu** [comment]
Of course:

```
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro/envs/2679/lib/python3.8/site-packages/kedro_datasets/databricks/managed_table_dataset.py:docstring of kedro_datasets.databricks.managed_table_dataset.ManagedTableDataSet:30:Unexpected indentation.
```

**Comment by stichbury** [comment]
@jmholzer @noklam I'm afraid I probably need your help to get this resolved, please. Sphinx is emitting the following:

```
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro/envs/2679/lib/python3.8/site-packages/kedro_datasets/databricks/managed_table_dataset.py:docstring of kedro_datasets.databricks.managed_table_dataset.ManagedTableDataSet:30:Unexpected indentation.
```

And I can't see a docstring at that location in the latest version of the databricks dataset.

I want to get these docs built into our set because we have a blog post to publish that should point to them. Even if they don't get into release 0.18.11 it would be sensible to have them on `latest`, which needs a `kedro-datasets` release to fix any problematic docstrings.

Please could you take a look at the code and see what is causing Sphinx to fail and maybe separately we can review how to avoid this in future by running a suitable linter over each new dataset that we add?

**Comment by stichbury** [comment]
Thank you @noklam and @jmholzer for getting this building!

**Comment by noklam** [comment]
@stichbury Not sure where is the best place to fit this documentation. I can go through the doc build process with you.

1. The build process only happens when `kedro` have a new release
2. `kedro` will pull latest `kedro-datasets`, which means any change in `kedro-datasets` will require a new release in `kedro-datasets` before it show up in our docs.

# How do you test docs change?
Kedro's GitPod is the best place to test it. In our `setup.py`, it points to the latest `kedro-datasets`. When you need to test changes on a feature branch, you need to update the reference to point to your `kedro-datasets` feature branch.

You can either open a PR on `kedro` or just start GitPod and do `make build-docs` to check if documentation is build successfully. If you have a local setup you can also use your local environment, but I found GitPod convenient for these short-lived isolated environment.

```diff
# setup.py
-         # "kedro-datasets[all]~=1.4.1",
+        "kedro-datasets[all]@git+https://github.com/kedro-org/kedro-plugins@fix/rst-indentation-managedtabledataset#subdirectory=kedro-datasets",
```

**Review by astrojuanlu — APPROVED** [review]
Docs build passed! 🎉 

- https://kedro--2679.org.readthedocs.build/en/2679/kedro_datasets.spark.SparkStreamingDataSet.html
- https://kedro--2679.org.readthedocs.build/en/2679/kedro_datasets.databricks.ManagedTableDataSet.html

**Comment by noklam** [comment]
Let's merge it then we can go with the release

**Pull request merged**